docs: fix linting errors in architecture components guide (#47350)
angular/angular: aio/content/guide/architecture-components.md

| Directives                                  | Details |
|:---                                         |:---     |

> | [`*ngFor`](guide/built-in-directives#ngFor) | An iterative, it tells Angular to stamp out one `<li>` per hero in the `heroes` list. |

Reviewer: Can we just make this a complete sentence and remove the colloquialism?
```suggestion
| [`*ngFor`](guide/built-in-directives#ngFor) | An *iterative*, which tells Angular to create one `<li>` per hero in the `heroes` list. |
```

> | [`*ngIf`](guide/built-in-directives#ngIf)   | A conditional, it includes the `HeroDetail` component only if a selected hero exists. |

Reviewer: Same idea as the previous:
```suggestion
| [`*ngIf`](guide/built-in-directives#ngIf)   | A *conditional*, which includes the `HeroDetail` component only if a selected hero exists. |
```
